src/clj/puppetlabs/services/jruby/jruby_puppet_schemas.clj

```diff
@@ -183,3 +182,39 @@
   "Schema for a clojure persistent map for the system environment"
   (schema/both EnvMap
     (schema/either PersistentArrayMap PersistentHashMap)))
+
+(defn event-type-requested?
+  [e]
+  (= :instance-requested (:type e)))
+
+(defn event-type-borrowed?
+  [e]
+  (= :instance-borrowed (:type e)))
+
+(defn event-type-returned?
+  [e]
+  (= :instance-returned (:type e)))
+
+(def JRubyEventAction
```

Comment:
Yeah.  Not in love with that name.  To address @nwolfe 's earlier question, it is definitely required, because for metrics and logging and other things we always want to be able to show *some* indication of what triggered the borrow/return event.

If "Reason" seem like a better name than "Action" to everyone, I would definitely be happy to change it.  @jpinsonault does that seem more clear to you?  Or, do you have other suggestions for the name?